File: src/structures/costumer.ts
import {Entity, PrimaryGeneratedColumn, Column, OneToMany} from "typeorm";
import Address from "./address";
import Phone from "./phone";
import Product from "./product";
import Order from "./order";

@Entity()
export default class Costumer {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string;

    @OneToMany(()=> Phone, phone => phone.costumer, {cascade: true, eager: true})
    phones: Phone[];

    @OneToMany(()=> Address, address => address.costumer, {cascade: true, eager: true})
    addresses: Address[];

    @OneToMany(() => Order, order => order.costumer)
    orders: Order[];

    @Column({default: true})
    active:boolean;

}

File: Structures/Models/Costumer.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System;

namespace Comies.Structures.Models
{
    public class Costumer : Entity
    {
        public int Id { get; set; }
        
        [Required(ErrorMessage="Ops! Você precisa informar um nome.")]
        public string Name { get; set; }
        public string Document { get; set; }
        public DateTime MemberSince { get; set; }
        public virtual IList<Address> Addresses { get; set; }
        public virtual IList<Phone> Phones { get; set; }
        public virtual IList<Order> Orders { get; set; }

    }
}